### PR TITLE
[SAGE-294] Search foundations update

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] align search component to foundations 

## Screenshots

|  Before  |  After  |
|--------|--------|
|search-before-before|search-after|

## Testing in `sage-lib`

Visit the search component views and verify alignment with foundations spec:
- [Rails](http://localhost:4000/pages/component/search)
- [React](http://localhost:4100/?path=/story/sage-search--default)

## Testing in `kajabi-products`
A part of the foundations Epic PR

## Related

Closes [SAGE-294](https://kajabi.atlassian.net/browse/SAGE-294)

### CODE REVIEW

File: packages/sage-assets/lib/stylesheets/components/_search.scss [hunk 1]
```diff
@@ -18,26 +18,23 @@ $-search-icon: "::before";
     z-index: 2;
     margin-left: sage-spacing(xs);
     margin-right: -1 * sage-spacing();
-    color: sage-color(grey, 500);
+    color: sage-color(charcoal, 400);
```
Comment (monicawheeler) [inline packages/sage-assets/lib/stylesheets/components/_search.scss:21]: Looks like this value should be:

```suggestion
    color: sage-color(grey, 600);
```

File: packages/sage-assets/lib/stylesheets/components/_search.scss [hunk 2]
```diff
@@ -58,7 +55,10 @@ $-search-icon: "::before";
 
     &:focus-within {
       z-index: sage-z-index(default, 2);
-      box-shadow: map-get($sage-toolbar-button-borders, focus);
+      border-radius: sage-border(radius);
+      // box-shadow: map-get($sage-toolbar-button-borders, focus);
```
Comment (monicawheeler) [inline packages/sage-assets/lib/stylesheets/components/_search.scss:59]: Do we need to keep this comment?

File: packages/sage-assets/lib/stylesheets/components/_search.scss [hunk 3]
```diff
@@ -128,7 +128,7 @@ $-search-icon: "::before";
   &:focus:not(:disabled),
   &:active:not(:disabled),
   &:valid:not(:placeholder-shown) {
-    box-shadow: none;
+    // box-shadow: none;
```
Comment (monicawheeler) [inline packages/sage-assets/lib/stylesheets/components/_search.scss:131]: Also, should this comment be removed?

**Review by monicawheeler — CHANGES_REQUESTED** [review]
A couple of notes to review!

**Review by anechol — CHANGES_REQUESTED** [review]
Nice work! Just a few changes:
- Can we update the disabled font color to `Grey 600`. The icon looks like it should be about the same color but it's listed as `Charcoal 400` in the Design files. Judging by the color system, it does seem to line up with what the current Grey color should be so the icon might be fine as is.
- You can barely see this, but can we fix the border spacing here to match with the background?
Screen Shot 2022-03-31 at 10 32 07 AM

General question: 
I was under the impression that we were deprecating floating labels? Is this no longer the case?

**Review by anechol — APPROVED** [review]
🚀

**Pull request merged**